src/users/users.service.ts

```diff
@@ -0,0 +1,54 @@
+import { PrismaService } from '../database/prisma.service';
+import {
+  ConflictException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+import { UserDto } from './dto/user';
+import { Prisma } from '@prisma/client';
+import { PrismaError } from '../../prisma/prisma-error.enum';
+
+@Injectable()
+export class UsersService {
+  constructor(private readonly prismaService: PrismaService) {}
+
+  async getByEmail(email: string) {
+    const user = await this.prismaService.user.findUnique({
+      where: {
+        email,
+      },
+    });
+    if (!user) {
+      throw new NotFoundException();
+    }
+    return user;
+  }
+
+  async getById(id: number) {
+    const user = await this.prismaService.user.findUnique({
+      where: {
+        id,
+      },
+    });
+    if (!user) {
+      throw new NotFoundException();
+    }
+    return user;
+  }
+
+  async create(user: UserDto) {
+    try {
+      return this.prismaService.user.create({
```

Comment:
Did you test if the `ConflictException` is thrown as expected? There is an `await` missing here